api/v1/clusterextensionrevision_types.go

```diff
@@ -43,23 +47,54 @@ const (
 )
 
 // ClusterExtensionRevisionSpec defines the desired state of ClusterExtensionRevision.
+//
+// A ClusterExtensionRevision represents a specific immutable snapshot of the objects
+// to be installed for a ClusterExtension. Each revision is rolled out in phases,
+// with objects organized by their Group-Kind into well-known phases such as namespaces,
+// rbac, crds, and deploy.
```

Comment:
This doc refers `ClusterExtensionRevision`, thus it should be placed on `type ClusterExtensionRevision struct`